.github/actions/init-pnpm/action.yaml

```diff
@@ -0,0 +1,30 @@
+name: init-pnpm
+description: 'Initialize the repo with pnpm and install all the dependencies'
+inputs:
+  node-version:
+    description: 'Node.js version'
+    required: true
+    default: 24.x
+runs:
+  using: composite
+  steps:
+    - name: Cache turbo build
+      uses: actions/cache@v5
+      with:
+        path: .turbo
+        key: ${{ runner.os }}-turbo-${{ github.sha }}
```

Comment:
```- name: Cache Turbo build outputs
  uses: actions/cache@v5
  with:
    path: .turbo
    key: ${{ runner.os }}-turbo-${{ hashFiles('**/package.json', '**/pnpm-lock.yaml') }}
    restore-keys: |
      ${{ runner.os }}-turbo-```
      
      [docs](https://docs.github.com/en/actions/reference/workflows-and-actions/dependency-caching#:~:text=with%3A%0A%20%20%20%20%20%20%20%20%20%20%23%20npm%20cache%20files%20are%20stored%20in%20%60~/.npm%60%20on%20Linux/macOS%0A%20%20%20%20%20%20%20%20%20%20path%3A%20~/.npm%0A%20%20%20%20%20%20%20%20%20%20key%3A%20%24%7B%7B%20runner.os%20%7D%7D%2Dbuild%2D%24%7B%7B%20env.cache%2Dname%20%7D%7D%2D%24%7B%7B%20hashFiles(%27**/package%2Dlock.json%27)%20%7D%7D)
      
   for your choice, its look better like here